velox/functions/prestosql/Arithmetic.h

```diff
@@ -84,25 +87,39 @@ struct IntervalMultiplyFunction {
   }
 
   template <
+      typename TResult,
       typename T1,
       typename T2,
       typename = std::enable_if_t<
           (std::is_same_v<T1, int64_t> && std::is_same_v<T2, double>) ||
-          (std::is_same_v<T1, double> && std::is_same_v<T2, int64_t>)>>
-  FOLLY_ALWAYS_INLINE void call(int64_t& result, T1 a, T2 b) {
+          (std::is_same_v<T1, double> && std::is_same_v<T2, int64_t>) ||
+          (std::is_same_v<T1, int32_t> && std::is_same_v<T2, double>) ||
+          (std::is_same_v<T1, double> && std::is_same_v<T2, int32_t>)>>
+  FOLLY_ALWAYS_INLINE void call(TResult& result, T1 a, T2 b) {
     double resultDouble;
     if constexpr (std::is_same_v<T1, double>) {
       resultDouble = sanitizeInput(a) * b;
     } else {
       resultDouble = sanitizeInput(b) * a;
     }
 
+    TResult min, max, maxResult;
+    if constexpr (std::is_same_v<TResult, int64_t>) {
+      min = kLongMin;
+      max = kLongMax;
+      maxResult = kMaxDoubleBelowInt64Max;
+    } else {
+      min = kIntegerMin;
+      max = kIntegerMax;
+      maxResult = kIntegerMax;
```

Comment:
Why don't we need a `kMaxDoubleBelowInt32Max` here?

Reply:
For `int64`, `kMaxDoubleBelowInt64Max` seems to be added to prevent compile errors:
```
/// Comparing double against int64:max() like the following would get compile
/// time error on some compilers(e.g. clang 12 and above):
```
I did not notice any errors when comparing `double` with `int32::max()`, unit tests are also added for `IntervalYm` product with `int32::max()` values. Please let me know if a `kMaxDoubleBelowInt32Max` should be used here.

Reply:
You don't need one. Double type can represent all 32 bit integers. I overlooked the double type.